File: HRMS.Core/Models/Notification.cs
using HRMS.Core.Models.Enums;
using System.Text.Json;

namespace HRMS.Core.Models;

public class Notification
{
    public NotificationType Type { get; set; } = NotificationType.Info;

    public Role Group { get; set; } = Role.Admin;

    public string Text { get; set; } = string.Empty;



    public string ToJson()
        => JsonSerializer.Serialize(this, new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
}

File: HRMS.Desktop/lib/models/notification.dart
import 'package:json_annotation/json_annotation.dart';

import 'enums/notification_type.dart';
import 'enums/role.dart';

part 'notification.g.dart';

@JsonSerializable()
class Notification {
  NotificationType type;
  Role group;
  String text;

  Notification(
    this.type,
    this.group,
    this.text,
  );

  factory Notification.fromJson(Map<String, dynamic> json) =>
      _$NotificationFromJson(json);

  Map<String, dynamic> toJson() => _$NotificationToJson(this);
}
